File: Architecture/TodoApp/android/event/src/main/java/com/novoda/event/DataObserver.java
package com.novoda.event;

import rx.Observer;

public abstract class DataObserver<T> implements Observer<T> {

    @Override
    public void onError(Throwable e) {
        throw new IllegalStateException("Data Pipeline failed, this should never happen", e);
    }

    @Override
    public void onCompleted() {
        throw new IllegalStateException("Data Pipeline completed, this should never happen");
    }

}

File: Architecture/UnidirectionalKotlinCats/core/src/main/kotlin/com/odai/architecturedemo/event/DataObserver.kt
package com.odai.architecturedemo.event

import rx.Observer

interface  DataObserver<T>: Observer<T> {

    override fun onError(p0: Throwable?) {
        throw UnsupportedOperationException("Error on event pipeline. This should never happen", p0)
    }

    override fun onCompleted() {
        throw UnsupportedOperationException("Completion on event pipeline. This should never happen")
    }

}
